### ISSUE TITLE
No Loading animation on the website.

### ISSUE THREAD

**Issue by Tanishk05** [ethereum/ethereum-org-website#13600]
### Describe the bug

When I had first opened the site and click on any of the button, lets say explore button on the home page it waits for sometimes and then loads the page, but during that time I was not able to tell whats happening whether it is processing or not or my internet is slow.

### To reproduce

1. Go to 'https://ethereum.org/en/'
2. Click on 'Explore Ethereum'
3. See error

### Expected behavior

Its expected behavior is to show some kind of loading animation to tell users that something is happening and the site is not stuck. 

### Screenshots

# See this video
https://github.com/user-attachments/assets/7c1fd5c2-e7e8-49fb-8058-53ee9788163f

### In this, when I click on the buttons or the links down it takes some time to take to the destination. 

Lets say there is person whose internet is not that good, he may feel like stuck so this issue needs to be fixed.

### Would you like to work on this issue?

- [X] Yes
- [ ] No

**Comment by wackerow** [comment]
Hey @Tanishk05, appreciate you noting your experience here. Few things here about updates already in progress:

- Performance improvements: the team currently has had a focus on identifying and solving performance issues especially for those with lower bandwidth connections, or weaker devices. A large part of this is the current migration to using [shadcn](https://ui.shadcn.com) using [tailwindcss](https://tailwindcss.com/), with the goal to offer improved loading times. @pettinarip leading these efforts
- Homepage revamp: another upgrade being built in parallel which will bring several changes to the homepage specifically. Through this, performance is also being kept in mind, and there has been discussion similar to the issue you've raised around minimizing the initial load time, potentially using more lazy loading of larger content (ie hero image) and using some css coloration/animation as a filler.

cc: @pettinarip Not sure if I missed it, but do we have another issue tracking the performance changes specifically? Can leave this issue open for now as a tracking item if we don't have anything better... 
@Tanishk05 I love that you expressed interest in working on this! As noted above it's a complicated issue that already has a few solutions in flight, so I'd say please feel free to add to the discussion here, or on [Discord](https://ethereum.org/discord), but in the meantime going to assign this to @pettinarip and myself.

You mentioned animations, did you have any specific thoughts on how you'd like to see the homepage be presented while loading?

**Comment by pettinarip** [comment]
> Hey @Tanishk05, appreciate you noting your experience here. Few things here about updates already in progress:
> 
> * Performance improvements: the team currently has had a focus on identifying and solving performance issues especially for those with lower bandwidth connections, or weaker devices. A large part of this is the current migration to using [shadcn](https://ui.shadcn.com) using [tailwindcss](https://tailwindcss.com/), with the goal to offer improved loading times. @pettinarip leading these efforts
> * Homepage revamp: another upgrade being built in parallel which will bring several changes to the homepage specifically. Through this, performance is also being kept in mind, and there has been discussion similar to the issue you've raised around minimizing the initial load time, potentially using more lazy loading of larger content (ie hero image) and using some css coloration/animation as a filler.
> 
> cc: @pettinarip Not sure if I missed it, but do we have another issue tracking the performance changes specifically? Can leave this issue open for now as a tracking item if we don't have anything better... @Tanishk05 I love that you expressed interest in working on this! As noted above it's a complicated issue that already has a few solutions in flight, so I'd say please feel free to add to the discussion here, or on [Discord](https://ethereum.org/discord), but in the meantime going to assign this to @pettinarip and myself.
> 
> You mentioned animations, did you have any specific thoughts on how you'd like to see the homepage be presented while loading?

There is no public issue to track performance issues / fixes at the moment.

IMO we should wait and see the end results of the shadcn & homepage migrations. If we still find that the UX is still bad for slow connections/devices, we should implement a change to the UI to make it obvious.

We discussed this before and one suggestion was to implement a progress bar at the top of the page for page transitions. This idea lets the user know about the lack of interactivity while the content is still visible and the native elements (links) still work from the first render.

**Comment by konopkja** [comment]
@wackerow @pettinarip i am closing this issue for now since we focused on performance a lot this year, let me know if we should keep it open
